src/linear_algebra/affine_space.lean

```diff
@@ -100,6 +102,183 @@ begin
   apply set.mem_of_mem_of_subset hp1p2 hp1p2s
 end
 
+section combination
+
+variables {k} {ι : Type*} [fintype ι]
+
+/-- A weighted sum of the results of subtracting a base point from the
+given points.  The main cases of interest are where the sum of the
+weights is 0, in which case the sum is independent of the choice of
+base point, and where the sum of the weights is 1, in which case the
+sum added to the base point is independent of the choice of base
+point. -/
+def weighted_vsub_of_point (w : ι → k) (p : ι → P) (b : P) : V := ∑ i, w i • (p i -ᵥ b)
+
+/-- The weighted sum when the weights are 0. -/
+@[simp] lemma weighted_vsub_of_point_zero (p : ι → P) (b : P) :
+  weighted_vsub_of_point V (0 : ι → k) p b = 0 :=
+begin
+  unfold weighted_vsub_of_point,
+  conv_lhs {
+    congr,
+    skip,
+    funext,
+    erw zero_smul
+  },
+  exact finset.sum_const_zero
+end
+
+/-- The weighted sum, multiplied by a constant. -/
+lemma weighted_vsub_of_point_smul (r : k) (w : ι → k) (p : ι → P) (b : P) :
+  r • weighted_vsub_of_point V w p b = weighted_vsub_of_point V (r • w) p b :=
+begin
+  erw finset.smul_sum,
+  congr,
+  ext,
+  rw ←mul_smul,
+  refl
+end
+
+/-- The weighted sum, negated. -/
+lemma weighted_vsub_of_point_neg (w : ι → k) (p : ι → P) (b : P) :
+  -weighted_vsub_of_point V w p b = weighted_vsub_of_point V (-w) p b :=
+begin
+  erw ←finset.sum_neg_distrib,
+  congr,
+  ext,
+  rw ←neg_smul,
+  refl
+end
+
+/-- Adding two weighted sums. -/
+lemma weighted_vsub_of_point_add (w1 w2 : ι → k) (p : ι → P) (b : P) :
+  weighted_vsub_of_point V w1 p b + weighted_vsub_of_point V w2 p b =
+    weighted_vsub_of_point V (w1 + w2) p b :=
```

Comment:
```suggestion
lemma weighted_vsub_of_point_add (w₁ w₂ : ι → k) (p : ι → P) (b : P) :
  weighted_vsub_of_point V w₁ p b + weighted_vsub_of_point V w₂ p b =
    weighted_vsub_of_point V (w₁ + w₂) p b :=
```
etc

Reply:
Made that change throughout for `w₁`, `w₂`, `b₁`, `b₂`.